### PR TITLE
Enable deadcode, goimports, misspell and prealloc linters

### PR DESCRIPTION
**Pull request by saschagrunert**
This enables the linters and fixes the corresponding reports.

### CODE REVIEW

**Comment by feiskyer** [comment]
Thanks to enable more linters. Could you check why the CIs are failing?

**Comment by saschagrunert** [comment]
> Thanks to enable more linters. Could you check why the CIs are failing?

I'm not sure and it looks like flakes. I did a rebase and will investigate further.

File: pkg/validate/image.go [hunk 1]
```diff
@@ -198,7 +198,7 @@ func testPullPublicImage(c internalapi.ImageManagerService, imageName string, po
 
 // pullImageList pulls the images listed in the imageList.
 func pullImageList(c internalapi.ImageManagerService, imageList []string, podConfig *runtimeapi.PodSandboxConfig) []string {
-	var ids []string
+	ids := []string{}
```
Comment (saschagrunert) [inline pkg/validate/image.go]: The issue was that the previous pre-allocation to `len(imageList)` compromised the test result. So this is fixed now. :)

Comment (feiskyer) [reply pkg/validate/image.go]: thanks for the fix

**Pull request merged**